content/src/service/snapshots/SnapshotManager.ts

```diff
@@ -35,18 +48,28 @@ export class SnapshotManager {
             (await this.deploymentsSince(entityType, snapshot.lastIncludedDeploymentTimestamp, transaction)) >=
               typeFrequency
           ) {
-            await this.generateSnapshot(entityType, transaction)
+            await this.generateSnapshotPerEntityType(entityType, transaction)
           }
         }
       },
       { priority: DB_REQUEST_PRIORITY.HIGH }
     )
   }
 
-  getSnapshotMetadata(entityType: EntityType): SnapshotMetadata | undefined {
+  async startCalculateFullSnapshots(): Promise<void> {
+    // TODO: Add metrics regarding snapshots
+    await this.generateSnapshot()
+    setTimeout(async () => await this.startCalculateFullSnapshots(), this.snapshotFrequencyInMilliSeconds)
```

Comment:
Use setInterval instead of setTimeout. Mind that `setInterval` does not accept async functions. You must catch the exception like this:

```diff
-setTimeout(async () => await this.startCalculateFullSnapshots(), this.snapshotFrequencyInMilliSeconds)
+setInterval(() => this.startCalculateFullSnapshots().catch(SnapshotManager.LOGGER.error), this.snapshotFrequencyInMilliSeconds)
```